src/utils/post_utils.js

```diff
@@ -175,3 +183,98 @@ export function comparePosts(a, b) {
 
     return 0;
 }
+
+export function combineSystemPost(postsIds = [], posts = {}) {
+    if (postsIds.length === 0) {
+        return {postsForChannel: postsIds, nextPosts: posts};
+    }
+
+    const postsForChannel = [];
+    const nextPosts = {...posts};
+
+    let userActivitySystemPosts = [];
+    let systemPostIds = [];
+    let createAt;
+    let combinedPostId;
+
+    postsIds.forEach((p, i) => {
+        const channelPost = posts[p];
+        if (
+            channelPost.delete_at === 0 &&
+            (isUserActivityPost(channelPost.type) || channelPost.type === Posts.POST_TYPES.COMBINED_USER_ACTIVITY)
+        ) {
+            if (!createAt || createAt > channelPost.create_at) {
+                createAt = channelPost.create_at;
+            }
+
+            if (isUserActivityPost(channelPost.type)) {
+                userActivitySystemPosts = [...userActivitySystemPosts, channelPost];
+                systemPostIds.push(channelPost.id);
+
+                if (nextPosts[channelPost.id]) {
+                    nextPosts[channelPost.id] = {...channelPost, state: Posts.POST_DELETED, delete_at: 1};
+                }
+            } else if (channelPost.type === Posts.POST_TYPES.COMBINED_USER_ACTIVITY) {
+                userActivitySystemPosts = [...userActivitySystemPosts, ...channelPost.user_activity_posts];
+                systemPostIds = [...systemPostIds, ...channelPost.system_post_ids];
+
+                combinedPostId = channelPost.id;
+            }
+        }
+
+        if (channelPost.type === '') {
+            if (userActivitySystemPosts.length > 0) {
```

Comment:
It would be nice if we could combine the two branches of this if statement since their logic is almost identical. Would this work if it was something like
```javascript
if (
    (channelPost.type === '' && userActivitySystemPosts.length > 0) ||
    userActivitySystemPosts.length === MAX_COMBINED_SYSTEM_POSTS ||
    (userActivitySystemPosts.length > 0 && i === postsIds.length - 1)
) {
    ...
}
```

Reply:
Done